exercise/templates/exercise/exercise_plain.html

```diff
@@ -80,7 +80,7 @@
 								<ul class="dropdown-menu">
 									{% for submission in submissions %}
 										<li>
-											<a href="{{ submission|url:'submission-plain' }}">
+											<a href="{{ submission|url:'submission-plain' }}" data-hash="{{ submission.meta_data.submission_hash }}">
```

Comment:
`exercise_plain.html` affects only exercises embedded in chapters. A+ also supports full-view exercises without chapters. Is the duplicate submission check supported in full-view exercises yet? If I remember correctly, the JS code in chapter.js also does not affect full-view exercises, in which case it could be better to write reusable JS functions outside chapter.js so that they can be called outside chapter.js too.

Full-view exercise html template:
https://github.com/apluslms/a-plus/blob/df64e846d83436bccd92e310662f449a4ff74210/exercise/templates/exercise/exercise_base.html#L59-L63

Reply:
I have now attempted to add the duplicate check to the full-view exercises too.

I couldn't get it to work properly, mainly because I would somehow need the full-view exercise pages to support the `AplusExercise` prototype from `chapter.js`, since I need to be able to check if the exercise is a quiz or active element exercise.

Do you have any ideas how to make this work or some workarounds?

Reply:
I wrote my answer in the review comment today:
https://github.com/apluslms/a-plus/pull/1048#pullrequestreview-1058583782